File: chapter16/echoFac.erl
-module(echoFac). 
-export([test/0]). 

test() -> 
  Cmd = "ruby echoFac.rb", 
  Port = open_port({spawn, Cmd}, [{packet, 4}, use_stdio, exit_status, binary]),
  Payload = term_to_binary({fac, list_to_binary(integer_to_list(23))}), 
  port_command(Port, Payload), 
  receive 
    {Port, {data, Data}} -> 
      {result, Text} = binary_to_term(Data), 
      Blah = binary_to_list(Text),
      io:format("~p~n", [Blah]) 
  end.

File: chapter16/echoFac.rb
require 'rubygems' 
require 'erlectricity' 
require 'stringio' 

def fac n
 if (n<=0) then 1 else n*(fac (n-1)) end
end

receive do |f| 
  f.when(:fac, String) do |text| 
    n = text.to_i
    f.send!(:result, "#{n}!=#{(fac n)}") 
    f.receive_loop 
  end 
end 



